Social Icons: Add initial e2e tests

Jagjeet opened the pull request:
#### Changes proposed in this Pull Request

* Adds e2e tests for testing social icons block

#### Testing instructions
* Run the tests using the command below: 

./node_modules/.bin/mocha specs/wp-calypso-gutenberg-social-icons-block-spec.js

Jagjeet committed:
Add initial social icons e2e test

Jagjeet committed:
Add initial social icons e2e test

worldomonation commented:
Thanks for the PR! I realize I may have been a bit pushy for a PR, but this way I can see what you were trying to do and test out the code myself locally to check why you'd be running into the issue that you were seeing.

jsnajdr committed:
MediaLibrary: remove unused onEditItem prop (#50829)

renovate[bot] committed:
chore(deps): update dependency webpack to ^5.24.4 (#50831)

* chore(deps): update dependency webpack to ^5.24.4


Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Sergio Cinos <sergio.cinos@gmail.com>

daledupreez committed:
Ensure that `errorNotice` is correctly wired into outbound Titan redirect component (#50707)

* Correctly wire in errorNotice
* Document additional props

Co-authored-by: Dale du Preez <dale@automattic.com>

daledupreez committed:
Show all errors stemming from addition to cart (#50700)

Co-authored-by: Dale du Preez <dale@automattic.com>

yansern committed:
Domain Suggestions: Externalize getSuggestionsVendor(). (#50756)

* Added getDomainSuggestionsVendor() in @automattic/domain-picker package and update getSuggestionsVendor() to reuse the function.

* Update comment.

Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

* Remove optional operator.

Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

* Fallback to config when isPremium prop is not given.

Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

scinos committed:
chore(packages): Update webpack packages (#50886)

* fix(deps): update dependency webpack-bundle-analyzer to ^4.4.0
* fix(deps): update dependency webpack-cli to ^4.5.0
* fix(deps): update dependency webpack-dev-middleware to ^4.1.0

Co-authored-by: Renovate Bot <bot@renovateapp.com>

keoshi committed:
Jetpack Connect: add busy state to the button while isFetching (#50849)

flootr committed:
js-utils: make package private, add code owners, add important notice, change version (#50766)

* remove `@automattic/js-utils` and move fns to `client/lib`

* Revert "remove `@automattic/js-utils` and move fns to `client/lib`"

This reverts commit bea3f7f1bf075d3aa1bcde801ae9b8c8c455c791.

* js-utils: change version to 0.1.0 and mark as private

* js-utils: mark @Automattic/team-calypso-frameworks as code owners

* js-utils: add a note about the purpose of this package

* Update packages/js-utils/README.md

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

* Update .github/CODEOWNERS

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

renovate[bot] committed:
fix(deps): update lodash monorepo to ^4.17.21 (#50885)

* fix(deps): update lodash monorepo to ^4.17.21


Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Sergio Cinos <sergio.cinos@gmail.com>

flootr committed:
people(team): use `react-query` to fetch data (#50712)

* people(users): use `useUsers` query to fetch data

* team: use `useTranslate` instead of `localize`

* useUsers: set better defaults and consider custom pagination param

* useUsers: use `wpcom.req.get` directly instead of `wpcom.site().usersList()`

* team: move `listKey` up & don't pass on `fetchOptions`

* team: extract `useErrorNotice`

* team: make expected props more explicit

* team: explicitly pass on props

* rename `useUsers` to `useUsersQuery`

getdave committed:
Decouple the Happy Chat button from the FAB and always show it when there’s an active chat session (#50800)

* Naive attempt to enable HC if there is active session

* Decouple Happy Chat button from Inline Help

* Removes unused dependency.

* Removes attemptStartHappyChat.

* Removes openChat action.

* Removes chast styles from inline-help.

* Minor comment fixing.

* Fixes classes, adds mobile redirect.

* Simplifies CSS by setting borderless:false.

* Hides FAB in home by adding home section in shouldLoadInlineHelp().

* Enables chat in jetpack-connect, removes /me/account/closed as it no longer exist.

* Adds the why for css offset.

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

scinos committed:
ci(test): Create an explicit browserslist env for tests (#50880)

scinos committed:
Revert "ci(test): Create an explicit browserslist env for tests (#50880)" (#50891)

This reverts commit 4a6b374c8ebd030fafca3be97cbd558cc1f689bd.

mmtr committed:
Nav Unification: Add param on external URLs for toggling the feature on WP Admin (#50839)

aneeshd16 committed:
Adds test to emphasize the free domain option in the launch flow (#50324)

* Adds test to emphasize the free domain option in the launch flow

* Adds Experiment component

* Updates experiment name

keoshi committed:
Activity Log: update more info to an external link (#50851)

sarayourfriend committed:
search: Convert to function component (#50004)

* search: Convert to function component

* Refactor with useImperativeHandle

* Only call onSearchProp after inital mounting

* Add `useInstanceId` to definition for wp/compose

* Remove unnecessary memoization and use `useState` instead of `useReducer`

* Remove errant prop type annotation

* Stop using FunctionComponent and company

* Support defaultValue searching

* Add testing-library dependencies

* Add tests for search

* Do not debounce initial search

* Deduplicate testing libraries

* Extract complex type into alias

* Further remove unnecessary memoizations

* Remove redundant null check

michaeldcain committed:
Hide refund text when purchse is free (#50855)

michaeldcain committed:
Cancellation: translate button text (#50856)

* Wrap button text in translation function

* Pull translate out of ternary

Commit:
Update styles of the asterisk notes in pricing page (#50900)

blackjackkent committed:
Update warning banner for custom nameservers to be hidden when the nameservers are Cloudflare's (#50862)

sixhours committed:
Buttons: Ensure pointer-events are disabled for busy buttons (#50906)

michaeldcain committed:
Codeowners: add View Design to block patterns (#50908)

Jagjeet commented:
No worries! I was past the point of just seeing how things work and needed to get to a point where I could share my code anyways. Just came sooner than planned with running into that issue. The PR wasn't too much on top of that.

niranjan-uma-shankar committed:
Signup domains step: Fix alignment for RTL (#50835)

* Fix alignment for RTL

* Fix border on the search box in RTL

Commit:
Fix home link accessible name in Jetpack masterbar (#50859)

* Fix accessibility issues in cloud Jetpack masterbar

* Fix TS issue

akirk committed:
Localize URL: Don't rewrite wordpress.com URLs that point to Calypso (#50897)

* Don't rewrite wordpress.com URLs that point to Calypso

* The URL needs to be the last path part

worldomonation committed:
Add new issue template for Task type (#50873)

- add a new template via the online GitHub issues template editor for the `Task` type.

sarayourfriend committed:
Components: Refactor Card to TypeScript (#50863)

sirbrillig committed:
Move getRememberedCoupon to checkout controller (#50817)

pablinos committed:
Nav Unification: Disable Calypsoify with the 'from' parameter (#50923)

Going to `wp-admin` on a site with Calypsoify enabled, turns it off.
With #50839 we added a `from` parameter, which prevented that from
happening.

This PR explicitly disables Calypsoify by adding the an additional query
parameter to the URL as we add the `from` parameter.

ramonjd committed:
Add fredrickson to available designs (#49970)

jsnajdr committed:
Activity Log: fix and test restore persistence (#50748)

* Activity Log: test restore persistence

* Activity Log: fix restore persistence

* Activity Log: remove restore persistence 😜

scinos committed:
ci(test): Create an explicit browserslist env for tests (Take 2) (#50898)

* Revert "Revert "ci(test): Create an explicit browserslist env for tests (#50880)" (#50891)"

This reverts commit 46ff6037fe3359943146911f99346c13e3de53ca.

* Fix snapshots

renovate[bot] committed:
fix(deps): update dependency lerna to v4 (#50887)

* fix(deps): update dependency lerna to v4
* Update `jsdom` and deduplicate packages

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Sergio Cinos <sergio.cinos@gmail.com>

worldomonation committed:
Remove references to IE11 from active e2e documentation. (#50864)

- supporting the deprecation of IE11, e2e documentation will remove references to IE11.

frontdevde committed:
Color Schemes: add back default (#50933)

* Color Schemes: add back default

* Change selector

cpapazoglou committed:
[DON'T MERGE] Launch nav-unification for all new users and 5% of existing users. (#50529)

* Adds nav-unification util functions.

* Hides the modal for new users.

* Correctly get 5% of users. Ids ending in 00,01,02,03,04.

* Rolls out to the current segment 24h after launch.

* Rebase and moves utils in selector.

* Updates selector to user max user id.

* Swithches to jsx.

* Reuses isNavUnificationNewUser function.

* Excludes test users.

* Uses isE2ETest instead of static User Ids.

* Extracts userInCurrentRolloutSegment in variable.

* Adds comment for why not showing modal to new users.

* Updates new user id.

* Updates comment.

* Updates comment.

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

Aurorum committed:
Stats: Allow Dismissing Earn Nudge with Some Plans (#46343)

* Stats: Allow Dismissing Earn Nudge with Business Plan

* Use CTA in banner

* Update site.jsx

* Remove exclamation mark

ddc22 committed:
Hide validation errors on lost focus of empty form (#50477)

sarayourfriend committed:
state-utils: Extract extendAction (#50124)

* state-utils: Extract extendAction

* Fix bad rebase

* Extract latest version

* export default extendAction

* Move to external deps section

* Update inline docs

* Sort exports

Co-authored-by: Marin Atanasov <tyxla@abv.bg>

klimeryk committed:
Domains: Ensure enough space for DNS record type (#50846)

Fixes #50823

Adjusts the min-width to ensure the DNS record type label does not clip.

mattgawarecki committed:
Plans: Add the page header from Calypso Green to Calypso Blue (#50920)

mattgawarecki committed:
Jetpack Backup: Render progress % on page reload (#50860)

jessie-ross committed:
Update the A/A slug (#50948)

alshakero committed:
Hide CTAs in success view when it isn't needed (#50737)

donlair committed:
Package version of What's New Guide component (#50224)

* updates

* updating WhatsNewGuide component

* Moving WhatsNew component to its own package.

* Updating component

* Linter fixes

* rebase

* Working on addressing feedback

* style updates

* style tweaks

* Addressing review feedback

* Package.json update

* Removing experimental Text component, adding translation context, and removing unused className check

* Removing component placement from Marketing Tools

* attempting to fix yarn.lock

* reverting yarn.lock

* Reverted yarn.lock and then ran yarn && yarn start

* Reverting yarn.lock

* Updating i18n domain

* Remainder of i18n updates

* Missed one save

* Addressing feedback and tweaking dependencies to reduce yarn.lock changes

* rebase

* trying to get build tests to pass yarn.lock changes

* more work on yarn

* Rebase and string update

* Rebase

roundhill committed:
VideoPress: Update media library preview to use new player (#49896)

* Update the media library video preview to use the new videojs player. Also updated proxy function to work properly with what new player expects (A getVideoInfo function passed that uses a callback)

* Lint fix.

mattgawarecki committed:
ABTest: Add social proof header to Jetpack pricing pages (#50947)

* Create new test and iteration for 'jetpackSocialProofHeader'

* Create 'useMaybeSocialProofHeader' hook to show social proof header

* Show social proof header (if appropriate) on Jetpack plan/pricing pages

* Fix abnormal whitespace character to re-use existing translations

* Update test start date

mattsherman committed:
Add Powered by WooCommerce footer to the WooCommerce installation page (#50726)

klimeryk committed:
Domains: Show the required root A records for mapping (#50668)

As explained in #50660, when mapping a domain, you get a nudge in Calypso to update the DNS records to point to our nameservers. However, it's also possible to just update the root A records. We mention this option in our support docs, but not in the UI.

This PR updates the mapped domain settings page to show the A records as an advanced setup option. The recommended configuration - changing the name servers to ours - is shown open by default, with the advanced setup closed.

mattsherman committed:
Remove everything from Store dashboard except for removal notice (#50677)

- Redirect to WooCommerce installation page from Store dashboard
- Convert dashboard component to a functional component
- Refactor controller to remove unused code paths

worldomonation committed:
Disable the IE11 canary configuration on CircleCI. (#50960)

noahtallen committed:
ETK: Remove old GH actions (#50878)

- Remove redundant test from GH actions.
- Refactor TeamCity test order to implicitly include test we were doing on newspack blocks.
- No longer run any step when the previous fails.

ramonjd committed:
Onboarding: Pollard should be using blank-canvas (#50975)

* Pollard should be using blank-canvas

* Balan needs blank-canvas too

jsnajdr committed:
Fix apiNamespace and apiVersion in JITM REST request (#50941)

renovate[bot] committed:
chore(deps): update dependency chokidar to v3 (#50937)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

scinos committed:
chore(lint): Fixes lint errors (#50935)

daledupreez committed:
Add feature flag for centralized/new email home page (#50977)

* Add feature flag
* Rename feature flag to `email/centralized-home`

Co-authored-by: Dale du Preez <dale@automattic.com>

daledupreez committed:
Update Titan control panel iframe height to 1200px (#49520)

* Update height to 1000px for testing
* Increase height to 1200px per Titan

Co-authored-by: Dale du Preez <dale@automattic.com>

stephanethomas committed:
Emails: Prevent placeholder from showing on Email page when site domains are fetching (#50177)

* Emails: Remove unnecessary indentation in stylesheets

* Emails: Prevent placeholder from showing when site domains are fetching

We should only show it when G Suite users and site domains has not loaded yet. This solves an issue where the placeholder would be displayed twice when the Email page is reloaded manually.

* Emails: Simplify imports of Titan functions

* Emails: Fix invalid translation comment in Add New Users page

WunderBart committed:
E2E: Don't run GB upgrade edge specs when non-edge part fails (#50942)

sixhours committed:
Add a custom subheading to the Testimonials post type header area.

cpapazoglou committed:
Fixes arrow position when menu is open. (#50932)

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

sarayourfriend committed:
Components: Turn Card into function component and forward refs (#50924)

* Turn Card into function component and forward refs

* Update tests for new Card component display name

* Use children as an array

* Remove ForwardRef from display name

sirbrillig committed:
Checkout: Convert useCreatePaymentMethods to TypeScript (#50965)

* Convert useCreatePaymentMethods to TypeScript

* Add type for StripeLoadingError

* Cast labelText for useCreatePayPal to string

* Make the object argument required by useCreatePayPal

* Fix type of activePayButtonText in useCreateCreditCard

kwight committed:
Anchor: Correct the Quick Link bump stats key. (#50963)

Commit:
Fix typo in activation error (#50968)

mattsherman committed:
Make sure plugin state is loaded before showing setup page (#51005)

brandonpayton committed:
Do not imply Happiness accepts ToS in support sessions (#50850)

jessie-ross committed:
ExPlat: Make development mode behaviour match production (#50995)

* Make development behaviour match production

* Fix tests, removing developmentMode behavior tests

jessie-ross committed:
Test original function calls (#50996)

yansern committed:
Domain Suggestions: Avoid vendor prop during store registration & types cleanup. (#50761)

* Remove vendor prop when registering domain suggestions data store. Fix up type issues on domain suggestions data stores.

* Hardcode vendor in domain store as it is overriden by domain picker. Fixes cyclic dependency issue.

* Remove noop.

* Remove domain picker dependency in data stores pacakge.

* Vendor will always have a default value. Additional cleanup.

* Move vendor assignment over to vendor prop instead of during registration of data store for Gutenboarding.

tyxla committed:
Media: Deselect only transient media items (#50982)

tyxla committed:
Components: Remove proptype limitation from ButtonGroup (#50984)

* Components: Remove proptype limitation from ButtonGroup

* Wrap in React.memo()

* Remove obsolete test

taggon committed:
Fix: feature list showing duplicated items (#50943)

ddc22 committed:
Fix: broken back function in domain only flow (#50743)

* Fixed and refactored getPreviousStep function to outright ignore the current step when considering the previous step

* Fixed and refactored getPreviousStep function to outright ignore the current step when considering the previous step

* Fixed unit tests

* Added unit tests for the previous step funftion

* Fixed issue with wasSkipped logic

Aurorum committed:
Media Library: Provide VideoPress Shortcode (#50895)

* Media Library: Provide VideoPress Shortcode

* Ensure "Copy" button isn't affected by longfade

mattgawarecki committed:
Remove lodash references from Calypso Green (#50458)

* Remove 'includes' from Jetpack Cloud landing component

* Remove 'get' from getJetpackCredentials selector

* Remove 'isEmpty' from Calypso Green settings

* Remove 'find' from Partner Portal license state filter

* Remove 'get' from Partner Portal partner key selector

* Remove 'isEmpty' from credentials wizard

* Remove 'memoize' from Partner Portal sidebar

* Remove 'get' and 'isArray' from Backup Successful card

* Remove 'noop' from Backup card

* Remove 'isNumber' from Jetpack product card

* Remove 'isNumber' from Scan badge

* Remove 'noop' from Scan threat item

* Remove 'get' and 'isEmpty' from server creds form

* Remove 'memoize' from Jetpack sidebar

daledupreez committed:
Allow Titan/Email management paths to be accessed in domain-only sites (#51036)

This change adds Titan/Email-related paths to the set of paths that can be accessed for domain-only sites.

Fixes #51025

sirbrillig committed:
Checkout: Remove CartStore from checkout (#50681)

* Remove CartData wrapper around CheckoutSystemDecider

* Remove cart store race condition guard in CheckoutSystemDecider

* Allow usePrepareProductsForCart to load products from localStorage

* Mock initial checkout cart request if no site/no user

* Prevent skipping usePrepareProductsForCart if product from url empty

* Fill in product_id for products from localStorage

* Correct return type of getCartFromLocalStorage

keoshi committed:
Remove stray character (#50985)

mattgawarecki committed:
Don't render the sidebar for /settings when no site is selected (#51032)

Commit:
Implement accessible profile dropdown (#50989)

* Implement accessible profile dropdown

* Update profile dropdown appearance

* Add label to user profile navigation landmark

* Add comments to style rules

* Prevents the Gravatar from moving to the right

* Simplify the scale transform and add it to focus as well

* Use typographic variables

Co-authored-by: Filipe Varela <keoshi@keoshi.com>

mattwiebe committed:
Site Editor: open published pages in site editor (#50866)

For already-published pages, edit them in the Site Editor rather than the Post Editor

Fixes #50702

donlair committed:
Swapping in new Facebook card content on Marketing Tools (#51038)

worldomonation commented on test/e2e/specs/wp-calypso-gutenberg-social-icons-block-spec.js:
```diff
@@ -0,0 +1,66 @@
+
+			return await driverHelper.waitTillPresentAndDisplayed(
+				driver,
+				By.css( '.wp-block-social-icons' )
```
Have you tried changing this selector to refer to 
`.wp-block-social-links`?

`driverHelper` should be able to find this element at that point, if the internal checks in `addBlock` are bypassed.

worldomonation replied:
For the actual `gEditorComponent.addBlock` method failing due to mismatch between the `Social Links` and `Social Icons` aria name, I added the following in the case-switch of function `getBlockSelectorSettings`:

```
			case 'Social Links':
				blockSettings = { ariaLabel: 'Social Icons'};
				break;
```

Jagjeet replied:
Thanks! It was failing in the addBlock function before that. Combined with the Aria change, I may be able to use 'Social Links' and then use .wp-block-social-links. I can test it out during the week.

worldomonation replied:
Right, as far as I understand the execution flow:
- `addBlock` function calls the function `getBlockSelectorSettings`.
- `getBlockSelectorSettings` looks for `Social Links`, does not find anything in the case switch, so returns the standard values.
- down the line in `addBlock` where validation occurs, it looks for a block with aria label `Social Links` which doesn't exist, as you have found - because its actually called `Social Icons`.

The addition of the case statement means the appropriate `ariaLabel` is returned to pass the validation. Ideally, the block name and ariaLabel should match, but as you can see in the other case statements unfortunately we have other blocks that are mismatched as well.

Jagjeet replied:
Thanks Edwin! Sorry about the delay. I tested this fix out and it works fine now. I should be able to move forward on these now.

jsnajdr committed:
Automated Transfer State: refactor query component, move tests (#50978)

* QueryAutomatedTransferEligibility: rewrite to functional component

* Automated Transfer Selectors: move tests to module directory, remove Chai

* QueryAutomatedTransferEligibility: remove unit test

* Fix props bug and re-add proptypes

jessie-ross committed:
Fix logstash properties (#51020)

jessie-ross committed:
ExPlat: Remove finished A/A experiments (#51017)

* Remove finished explat experiments

* Remove import

jessie-ross committed:
ExPlat: Implement ProvideExperimentData (#50993)

* Implement ProvideExperimentData

* FIx formatting

* Have useExperiment state change atomically

jessie-ross committed:
Use slot form for Experiment component (#50997)

* Use slot form for Experiment component

* Rename slots, append with "Experience"

* Fix formatting

aneeshd16 committed:
Updates "included in paid plans" text for domain transfer (#50837)

jsnajdr committed:
Atomic Transfer State: remove persistence and other complexity (#50936)

* Atomic Transfer State: remove persistence and other complexity

* CheckoutThankYou needs export for tests, guard for fetchTransferStatus

* Simplify mapDispatchToProps even more

* CheckoutThankYouHeader: add key prop to li items

jsnajdr committed:
Comments: remove extendAction usage (#50940)

bperson committed:
Analytics: Load latest `w.js` (#50904)

Fix fetching multiple times the same pixels in some circumstances ( high latency
connection, pixels recorded before `w.js` was loaded ). Those dups were
discarded on the backend, but there is no need to have the extra traffic ;)

This also switches back to `w.js` since `w2.js` was a way of deploying the new
Tracks client only in Calypso, now that `w.js` and `w2.js` are the same file,
there is no need to keep both of them around.

tyxla committed:
Framework: Remove (almost all of) tinymce's usage 🎉 (#51011)

* WooCommerce: Remove unused CompactTinymce component

* Guided Tours: Move wpcom-track-paste sources to a library

* Components: Remove tinymce and its obsolete references

* Server: Remove static files serving

* Static: Remove unused tinymce skins

* Build: Remove tinymce globals special treatment

* Client: Remove tinymce dependency 🎉

tyxla committed:
WooCommerce: Delete unused components (#51012)

tyxla committed:
Framework: Refactor away from _.castArray() (#51026)

* Framework: Refactor away from _.castArray()

* ESLint: Deprecate _.castArray()

WunderBart committed:
Update chromedriver to 89.0.0 (#50986)

* Update chromedriver package to v89.0.0

* Add chromedriver logs to .gitignore

tyxla committed:
Framework: Refactor away from _.takeRight() (#51028)

* Framework: Refactor away from _.takeRight()

* ESLint: Deprecate _.takeRight()

unDemian committed:
Mark as seen on open post notification (#49983)

* Mark as seen on open post notification

* Update naming

* Fix linter

keoshi committed:
Update Color Studio version to 2.5.0 (#51030)

* Update Color Studio version to 2.5.0

Reference: https://github.com/Automattic/color-studio/commit/5623403c5350d02ee453c31c9ddec2dd3907a8db

* Update yarn.lock

unDemian committed:
Remove seen state from some sections (#50424)

* Remove seen state from some sections

* Fix post menu options

stephanethomas committed:
G Suite: Show annual and monthly price on Email Comparison page (#50983)

* G Suite: Show annual and monthly price on Email Comparison page

* G Suite: Simplify code in Email Comparison page

Commit:
Add unit tests to Jetpack Free card button (#51008)

* Compute Jetpack Free button props in hook

* Carry over `plan` query param

* Test Jetpack Free card button

Commit:
Implement login link button with an anchor tag (#51045)

flootr committed:
people(viewers): improve query and ViewersList (#50890)

* useViewers: use `wpcom.req.get` directly

* rename `useViewers` to `useViewersQuery`

* useViewersQuery: make `siteId` arg explicit

* viewers: extract `useErrorNotice`

* useViewersQuery: simplify defaults

* useViewersQuery: : change query params order

* ViewersList: make sure strings are translated in useErrorNotice

* ViewersList: add missing `dispatch` in `useErrorNotice`

rcanepa committed:
Jetpack Cloud: re-style the navigation bar according to Jetpack.com's latest changes (#51050)

* Replace div with nav for better accessibility

* Decrease Jetpack Logo size

* Remove condition that removed the 'Menu' label

* Replace 'Product Tour' with 'Security'

* Update CSS to make it look like Jetpack.com's bar

* Update Security item href

noahtallen committed:
ETK: Use artifact dependency from API (#50966)

* Use artifact dependency from TC API

* Try to add build number grep

noahtallen committed:
Documentation typo fix (#51070)

worldomonation committed:
Remove task definitions from CircleCI configuration. (#51044)

* Remove disabled configurations from file. Also eliminate the Safari canary task as well.

* Remove the `wait-calypso-live` configuration items.

sirbrillig committed:
Shopping Cart: Remove purchaseDomain from ResponseCartProduct (#51015)

* Remove properties from ResponseCartProductExtra that do not exist

* Add purchaseId to RequestCartProductExtra only

I can't be sure that it is used by the backend, but it might be for some
products.

* Remove uses of purchaseDomain in checkout

* Alter getThankYouPageUrl to remove purchaseDomain

* Remove all other uses of purchaseDomain

* Remove purchaseDomain from getRenewalItemFromCartItem

* Remove includedDomain from calypso when used for cart items

Note that this property still exists on purchase items, but that's a
very different object.

* Add missing previousRoute dependency to useGetThankYouUrl

* Update getThankYouPageUrl renewal test to include subscription_id

* Prevent getThankYouPageUrl from getting purchase url with missing data

sirbrillig committed:
Checkout: Move react to devDependencies of composite-checkout and shopping-cart (#51072)

* Move react/react-dom to devDependencies for composite-checkout

* Change Checkout children prop to be ReactNode

* Move react to devDependencies for shopping-cart

worldomonation committed:
Add playwright and dependencies to test/e2e and instruct playwright to install browser binaries into node_modules folder. (#50774)

* Add playwright and dependencies to package.json and yarn.lock files.

* Run deduplicate

* In the environment setup phase, export environmet variable to install browser binaries into node_modules folder.

* Install Playwright to node_modules and re-export the environment variable in the test run step. Then try running a demo script

* Modify demo test to:
- run in headful mode
- load the login page
- set the default timeout

* Refactor the Playwright demo.

* Remove the demonstration script from PR.

* Revert changes to the test file filter.

* Add in PLAYWRIGHT_SKIP_DOWNLOAD check to calypso-doctor, and set the aforementioned environment variable where appropriate in Dockerfiles.

donlair committed:
Changing logo partner to Logotron (#51049)

Jagjeet committed:
Merge branch 'add/social-icons-block-e2e-tests' of github.com:Jagjeet/wp-calypso into add/social-icons-block-e2e-tests

Jagjeet committed:
Add patch to look for correct aria label

Jagjeet committed:
Add e2e test for inserting social icons block

This adds a basic test of testing an out-of-box social icons block and
verifying it can be inserted and viewed in the previewed post.

Jagjeet committed:
Added a first set of Social Icons Block testing

This included tests for the following:
 * Out of box configuration
 * Configuring social icons and testing they were published correctly
 * Editing social icons and testing they were published correctly
 * Configuring and deleting social icons and testing they were published correctly
 * Configuring, resizing and realigning social icons and testing they were published correctly

Jagjeet commented:
@worldomonation - I went ahead and finished this up since I was already ramped up on it. I had a few questions: 

- Can you give it a look and let me know if I should make any further changes? 
- When I rebased, it seemed to pull in a lot of other reviewers and changes. Is that normal? Is it okay to mark Ready for review as is?

Thanks!

worldomonation commented:
Thanks for the PR!

>     * Can you give it a look and let me know if I should make any further changes?

Sure thing, will take a look in the next coupe of days.

>     * When I rebased, it seemed to pull in a lot of other reviewers and changes. Is that normal? Is it okay to mark Ready for review as is?

When you incorporate the recent changes from `trunk`, my experience has been that I need to do a `git rebase trunk` instead of `git merge trunk` (this both assumes you're on the development branch). Otherwise what you're seeing happens where all work done on `trunk` gets its own commit and pollutes the history.

Edit: I see you mention that you rebased. I don't know what might have caused this, but if you haven't done any more development past what you have pushed up, I would suggest trying to to back to the pre-rebased state if possible and correct the problem.

Could you try doing a rebase instead and force push the branch up?

Jagjeet commented:
> Sure thing, will take a look in the next coupe of days.

In looking at some of the other blocks, I realized I need to add code to make sure I am looking at the correct block in some of the selectors. (eg. add ${ this.blockID }). I am going to add that in, but not sure how quickly I will get to it.

> > * When I rebased, it seemed to pull in a lot of other reviewers and changes. Is that normal? Is it okay to mark Ready for review as is?
> > ```
> 
> When you incorporate the recent changes from `trunk`, my experience has been that I need to do a `git rebase trunk` instead of `git merge trunk` (this both assumes you're on the development branch). Otherwise what you're seeing happens where all work done on `trunk` gets its own commit and pollutes the history.
> 
> Edit: I see you mention that you rebased. I don't know what might have caused this, but if you haven't done any more development past what you have pushed up, I would suggest trying to to back to the pre-rebased state if possible and correct the problem.
> 
> Could you try doing a rebase instead and force push the branch up?

Thanks for the advice! I'll see if I can fix it that way.
